app/models/bulk_download.rb

```diff
@@ -25,4 +31,159 @@ def convert_params_to_json
   def self.viewable(user)
     user.bulk_downloads
   end
+
+  def validate_access_token(access_token)
+    return self.access_token.present? && self.access_token == access_token
+  end
+
+  def output_file_presigned_url
+    begin
+      return S3_PRESIGNER.presigned_url(:get_object,
+                                        bucket: ENV['SAMPLES_BUCKET_NAME'],
+                                        key: download_output_key,
+                                        expires_in: OUTPUT_DOWNLOAD_EXPIRATION).to_s
+    rescue StandardError => e
+      LogUtil.log_err_and_airbrake("BulkDownloadPresignError: #{e.inspect}")
+    end
+    nil
+  end
+
+  def server_host
+    ENV["SERVER_DOMAIN"]
+  end
+
+  # The Rails success url that the s3_tar_writer task can ping once it succeeds.
+  def success_url
+    return nil if server_host.blank?
+    "#{server_host}#{bulk_downloads_success_path(access_token: access_token, id: id)}"
+  end
+
+  # The Rails error url that the s3_tar_writer task can ping if it fails.
+  def error_url
+    return nil if server_host.blank?
+    "#{server_host}#{bulk_downloads_error_path(access_token: access_token, id: id)}"
+  end
+
+  # The Rails progress url that the s3_tar_writer task can ping to update progress.
+  def progress_url
+    return nil if server_host.blank?
+    "#{server_host}#{bulk_downloads_progress_path(access_token: access_token, id: id)}"
+  end
+
+  # Returned as an array of strings
+  def aegea_ecs_submit_command(
+    shell_command,
+    task_role: "idseq-downloads-#{Rails.env}",
+    ecr_image: "idseq-s3-tar-writer:latest",
+    fargate_cpu: "4096",
+    fargate_memory: "8192"
+  )
+    config_ecr_image = get_app_config(AppConfig::S3_TAR_WRITER_SERVICE_ECR_IMAGE)
+    unless config_ecr_image.nil?
+      ecr_image = config_ecr_image
+    end
+    shell_command_escaped = shell_command.map { |s| Shellwords.escape(s) }.join(" ")
+    command = ["aegea", "ecs", "run"]
+    command += ["--command=#{shell_command_escaped}"]
+    command += ["--task-role", task_role]
+    command += ["--ecr-image", ecr_image]
+    command += ["--fargate-cpu", fargate_cpu]
+    command += ["--fargate-memory", fargate_memory]
+    command
+  end
+
+  # Returned as an array of strings
+  def s3_tar_writer_command(
+    src_urls,
+    tar_names,
+    dest_url,
+    success_url: self.success_url,
+    error_url: self.error_url,
+    progress_url: self.progress_url
+  )
+    command = ["python", "s3_tar_writer.py"]
+    command += ["--src-urls"]
+    command += src_urls
+    command += ["--tar-names"]
+    command += tar_names
+    command += ["--dest-url", dest_url]
```

Comment:
In this case, there is an easy way to do that without using array concatenation:
```ruby
command = ["python", "s3_tar_writer.py",
                      "--src-urls", *src_urls,
                      "--tar-names", *tar_names,
                      "--dest-url", *dest_url]
```